### PR TITLE
DEV/MNT: migrate to latest standards with pyproject.toml [LCLSPC-603]

### PR DESCRIPTION
**Pull request by klauer**
## Description
* Upgrading by way of https://github.com/pcdshub/pcds-python-migration-tools
* pcdscalc has been migrated to use setuptools-scm, replacing versioneer, as its version-string management tool of choice.
* pcdscalc has been migrated to use the modern ``pyproject.toml``, replacing ``setup.py`` and related files.
* Older language features and syntax found in the repository have been updated to Python 3.9+ standards by way of ``pyupgrade``.
* Sphinx 6.0 is now supported for documentation building. ``docs-versions-menu`` replaces ``doctr-versions-menu`` and ``doctr`` usage for documentation deployment on GitHub Actions.  The deployment key is now no longer required.
* `pyupgrade` was used to update annotations, though there might not have been many here
* `pycln` was used to clean up imports
* Adjusts requirements pins to fix CI (https://github.com/pcdshub/pcdscalc/issues/21)

## Motivation and Context
* Migrating because we want to migrate our projects from Travis CI for one
* Requirements pinning issue: https://github.com/pcdshub/pcdscalc/issues/21

## How Has This Been Tested?
* GitHub Actions

## Where Has This Been Documented?
This PR and Jira [LCLSPC-603](https://jira.slac.stanford.edu/browse/LCLSPC-603)

### CODE REVIEW

**Review by tangkong — APPROVED** [review]
lgtm

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
 numpy
 scipy
-xraydb
+xraydb<4.5.0
+sqlalchemy<2.0
```
Comment (ZLLentz) [inline requirements.txt:4]: This looks like an unfortunate story

Comment (klauer) [reply requirements.txt:4]: https://github.com/pcdshub/pcdscalc/issues/21

Comment (ZLLentz) [reply requirements.txt:4]: Matches up with what we're running on pcds-envs

**Review by ZLLentz — APPROVED** [review]
lgtm

**Pull request merged**